### PR TITLE
Introduce a BP Search Block as a BuddyPress Add-on

### PR DESCRIPTION
**Pull request by imath**
We've decided during the 10.0.0 development cycle that new BuddyPress blocks would be packaged as BuddyPress Add-ons and published into the WordPress Blocks/Plugins directory.

Here's the first block as BuddyPress add-on: BP Search Block. It's a block that uses a very old feature that was available when the BP Default theme was our main front-end playground. A search form with search options to look for members, groups, blogs or posts. Here's how it looks into a regular WordPress page:

front-end

As you can see, I've added the possibility to search for Activities. BuddyPress doesn't support this for now, but I'll open a ticket to quickly add this support into 10.0.0. I believe being able to search in activities from this block is a must have!

Here's how it looks into the Blocks editor:

block-editor

From the Sidebar settings, you can select which component can be listed into the "Search for" radio group. Using this radio group, you can switch the default selected radio item.

This block requires WordPress 5.8 because it uses the [block.json Metadata](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/). It also uses version 2 of the Block API.

I've also adapted the BP Blocks plugin, so that it's easier to build these "standalone" blocks in the future.

My idea is to request a WordPress.org repository for this block so that it can be listed into the BuddyPress Add-ons tab we've added to the Plugins/Add new Administration screen.

### CODE REVIEW

File: add-ons/bp-search-block/block.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
+{
+	"$schema": "https://schemas.wp.org/trunk/block.json",
+	"apiVersion": 2,
+	"name": "bp/search-form",
+	"title": "Community Search",
```
Comment (renatonascalves) [inline add-ons/bp-search-block/block.json:5]: Missing `i18n`

File: add-ons/bp-search-block/block.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	"category": "buddypress",
+	"icon": "search",
+	"description": "A Block to search for posts, sites, activities, members or groups from any post, page or widget of your BuddyPress powered community site!",
```
Comment (renatonascalves) [inline add-ons/bp-search-block/block.json:8]: Missing `i18n`

File: add-ons/bp-search-block/bp-search-block.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * A BuddyPress Block to search for activities, members, groups, sites, posts from any post, page or widget of your community site!
```
Comment (renatonascalves) [inline add-ons/bp-search-block/bp-search-block.php:3]: > posts from any post?

File: src/add-ons/bp-search-block/block.json [hunk 4]
```diff
@@ -0,0 +1,57 @@
+{
+	"$schema": "https://schemas.wp.org/trunk/block.json",
+	"apiVersion": 2,
+	"name": "bp/search-form",
+	"title": "Community Search",
```
Comment (renatonascalves) [inline src/add-ons/bp-search-block/block.json:5]: Missing `i18n`

File: src/add-ons/bp-search-block/block.json [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+	"category": "buddypress",
+	"icon": "search",
+	"description": "A Block to search for posts, sites, activities, members or groups from any post, page or widget of your BuddyPress powered community site!",
```
Comment (renatonascalves) [inline src/add-ons/bp-search-block/block.json:8]: Missing `i18n`

**Review by renatonascalves — APPROVED** [review]
Looks good!

File: add-ons/bp-search-block/block.json (hunk 1 above)
Comment (imath) [reply add-ons/bp-search-block/block.json:5]: According to the documentation, this is automatically done, see https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#internationalization Moreover as it's a JSON file, I'm unsure i18n functions would behave as expected. I've tested with the fr_FR language, and I saw these props were included into the generated pot file by WP CLI but were not translated. That's why I'm overriding these props within the main block file (index.js).

File: add-ons/bp-search-block/block.json (hunk 2 above)
Comment (imath) [reply add-ons/bp-search-block/block.json:8]: See: https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#internationalization

File: add-ons/bp-search-block/bp-search-block.php (hunk 3 above)
Comment (imath) [reply add-ons/bp-search-block/bp-search-block.php:3]: Well yes 😄 , it's actually what can happen if the block is inserted into a post. I'll replace by "content"

File: src/add-ons/bp-search-block/block.json (hunk 4 above)
Comment (imath) [reply src/add-ons/bp-search-block/block.json:5]: See https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#internationalization

File: src/add-ons/bp-search-block/block.json (hunk 5 above)
Comment (imath) [reply src/add-ons/bp-search-block/block.json:8]: See https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#internationalization

**Comment by imath** [comment]
Thanks a lot for your feedbacks @renatonascalves 😍 . I'll improve the "posts in post" thing 👍

**Pull request merged**